icu4c/source/common/rbbiscan.cpp

```diff
@@ -829,16 +829,14 @@ static const UChar      chRParen    = 0x29;
 UnicodeString RBBIRuleScanner::stripRules(const UnicodeString &rules) {
     UnicodeString strippedRules;
     int32_t rulesLength = rules.length();
-    bool skippingSpaces = false;
 
     for (int32_t idx=0; idx<rulesLength; idx = rules.moveIndex32(idx, 1)) {
         UChar32 cp = rules.char32At(idx);
         bool whiteSpace = u_hasBinaryProperty(cp, UCHAR_PATTERN_WHITE_SPACE);
-        if (skippingSpaces && whiteSpace) {
+        if (whiteSpace) {
```

Comment:
This will fail if the rules contain quoted spaces.
The original code can also fail, although in fewer cases - it keeps single spaces intact, just deletes the second through the end of runs.
I suggest opening a new bug for this; I don't think there's a quick and easy fix.

Reply:
[I haven't finished reviewing yet, but need to stop for the moment.]

Reply:
https://unicode-org.atlassian.net/browse/ICU-21093

Reply:
Are we ok with potentially failing in more cases, even if unusual?
If so, we should remember to note this in the ICU 68 migration issues. (We should note the data file format change anyway.)